noir-projects/aztec-nr/aztec/src/note/note_emission.nr

```diff
@@ -5,11 +5,12 @@
 pub struct NoteEmission<Note> {
     pub note: Note,
     pub storage_slot: Field,
+    pub note_hash_counter: u32, // a note_hash_counter of 0 means non-transient
```

Comment:
NoteEmission was naturally a convenient container for the values needed for emission which were originally in a header.